site/_posts/2021-09-16-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1087 @@
+
+### Configure the folders and packages
+
+Now, you need to add a package.json in your `server` folder. Do that by navigating to the `server` folder and running
+
+`npm init -y`
+
+In your `server/package.json`, add a rule that tells our app where to find our Express server.
+
+```js
+...
+"scripts": {
+  "start": "node index.js"
+},
+...
+```
+
+Then, install your dependencies.
+
+```
+npm install express cors express-session request
+```
+
+What did you just install?
+- `cors`, so that you can avoid CORS errors.
+- `express`, so that you can use the Express package.
+- `express-session`, so that you can save data in a server-side session.
+- `request`, so that you can format HTTP requests in an organized fashion
+
+Your completed `package.json` should look like this:
+
+```
+{
+  "name": "server",
+  "version": "1.0.0",
+  "description": "",
+  "main": "index.js",
+  "scripts": {
+    "start": "node index.js"
+  },
+  "keywords": [],
+  "author": "",
+  "license": "ISC",
+  "dependencies": {
+    "cors": "^2.8.5",
+    "express": "^4.17.1",
+    "express-session": "^1.17.2",
+    "request": "^2.88.2"
+  }
+}
+```
+
+### Write the Express server
+
+We need to create some routes for our Express server to use.  
+
+Create a `routes` folder in `reactauthapp/server`
+
+```bash
+mkdir reactauthapp/server/routes
+```
+
+Create a file called `user.js` in that folder
+
+```bash
+touch reactauthapp/server/routes/user.js
+```
+
+Now, open that new file and create a `user` route in `server/routes/user.js`:
+
+```js
+// server/routes/user.js
+
+const express = require('express');
+const router = express.Router();
+const config = require('../../client/src/config');
+
+router.get('/', (req, res) => {
+  res.send({
+    user: {
+      email: 'test@fusionauth.io'
+    }
+  });
+});
+
+module.exports = router;
+```
+
+Why is this in an object? The reason is FusionAuth will ultimately send us the user object when a user logs in. We are starting out by using mock data. Our server will send the email address, and then our react application can display it.
+
+* A note about the user object: the user object has several properties that are always present. Email is one of them. For more information on what sorts of data you can get on a user, check out our [user API docs](/docs/v1/tech/apis/users/#retrieve-a-user).
+
+Next, in `server/index.js` let's import & use express with `const app = express()`, define our first route, `/user`, and then start our server. Remember, you're using port 3001 because port 3000 has been automatically taken by `create-react-app`.  We will add `cors` a little later on after we discover why we need it!
```

Comment:
```suggestion
Next, in `server/index.js` let's import and use `express` with `const app = express()`, define our first route, `/user`, and then start our server. Remember, you're using port 3001 because port 3000 has been automatically taken by `create-react-app`.  We will add `cors` a little later on after we discover why we need it!
```

Reply:
Thanks for explaining why we are using 3001! Would be great to have this up where it is first mentioned.